pymeasure/display/inputs.py

```diff
@@ -240,17 +240,27 @@ def validate(self, text, pos):
     def fixCase(self, text):
         self.lineEdit().setText(text.toLower())
 
+    def toDouble(self, string):
+        value, success = self.validator.locale().toDouble(string)
+        if not success:
+            raise ValueError('String could not be converted to a double')
+        else:
+            return value
+
+    def toString(self, value, format='g', precision=6):
+        return self.validator.locale().toString(value, format, precision)
```

Comment:
Just wondering: is there a specific reason to use the validator's locale, rather than the scientific input's locale (i.e., `self.locale()`)?

(same question of course for the toDouble)

Reply:
I actually was not aware that `ScientificInput` itself (and all `QWidgets`?) have a locale. If it is guaranteed, that the validator has the same locale, then the other one can also be used. If there is some chance that they are different, using the locale of the validator, that sanitizes the input text, would be more fitting.

Reply:
I'm not sure if there is a chance that the validator has a different locale... It makes sense indeed to ensure that we use the same as the validator, so let's just leave it as it is.